File: GridView/RowDetailsInGridView/RowDetailsInGridViewVB/RowDetailsViewDefinition.vb
Imports Telerik.WinControls.UI

Public Class RowDetailsViewDefinition
    Inherits TableViewDefinition
    Public Overrides Function CreateViewUIElement(ByVal viewInfo As GridViewInfo) As IRowView
        Dim tableElement As GridTableElement = CType(MyBase.CreateViewUIElement(viewInfo), GridTableElement)
        tableElement.ViewElement.RowLayout = CreateRowLayout()
        Return tableElement
    End Function

    Public Overrides Function CreateRowLayout() As IGridRowLayout
        Return New RowDetailsLayout()
    End Function
End Class

File: GridView/RowDetailsInGridView/RowDetailsInGridView/RowDetailsViewDefinition.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Telerik.WinControls.UI;

namespace RowDetailsInGridView
{
    public class RowDetailsViewDefinition : TableViewDefinition
    {
        public override IRowView CreateViewUIElement(GridViewInfo viewInfo)
        {
            GridTableElement tableElement = (GridTableElement)base.CreateViewUIElement(viewInfo);
            tableElement.ViewElement.RowLayout = CreateRowLayout();
            return tableElement;
        }

        public override IGridRowLayout CreateRowLayout()
        {
            return new RowDetailsLayout();
        }
    }
}
